app/gui/src/model/suggestion_database/entry.rs

```diff
@@ -233,8 +270,29 @@ impl Entry {
     }
 
     /// Get the full qualified name of the entry.
-    pub fn qualified_name(&self) -> tp::QualifiedName {
-        tp::QualifiedName::new_module_member(self.module.clone(), self.name.clone())
+    pub fn qualified_name(&self) -> QualifiedName {
+        fn chain_iter_and_entry_name<'a>(
+            iter: impl IntoIterator<Item = &'a str>,
+            entry: &'a Entry,
+        ) -> impl Iterator<Item = &'a str> {
+            iter.into_iter().chain(iter::once(entry.name.as_str()))
+        }
```

Comment:
As previously, what is the advantage of embedding this function inside of the other function and making the other function longer and harder to read?